draft-dekater-scion-dataplane.md

```diff
@@ -1855,22 +1855,32 @@ Tunneling IP traffic over SCION requires a pair of SIGs and it involves the foll
+5. The egress SIG receives the SCION packet and decapsulates the original IP packet. It then forwards the packet to the final IP destination using standard IP routing.
 
 ## SIG Framing
 
-IP packets are encapsulated over SCION/UDP into SIG frames. Whilst in principle, a pair of SIGs may use other tunneling protocols, existing deployments use SIG framing as described here. This is to provide independence from the underlying SCION path MTU which can increase and decrease over time, to provide fast detection of packet loss and subsequent recovery of decapsulation for packets that weren't lost, and support for multiple streams within a framing session such that the streams can be distributed to separate cores.
+IP packets are encapsulated over SCION/UDP into SIG frames. Whilst in principle, a given pair of SIGs may use any tunneling protocol, existing deployments use the SIG framing as described here. This protocol is designed to:
+
+- provide independence from the underlying SCION path MTU which can increase and decrease over time.
+- provide fast detection of packet loss and subsequent recovery of decapsulation for packets that weren't lost.
+- support for multiple streams within a framing session such that independent packet sequences be tunneled in parallel.
 
 There may be multiple IP packets in a single SIG frame, and a single IP packet may be split into multiple SIG frames.
-A source SIG unidirectionally establishes a SIG tunneling session with a destination SIG. A session may be mapped to one or multiple SCION paths, depending on implementors.
 
-Streams can be used within a session to distinguish between different packet flows. A single stream has a single reassembly queue and SHOULD be processed by a single core. Packets in a stream SHOULD NOT be distributed over multiple SCION paths to avoid reordering and therefore lower performance.
+The ingress SIG initiates unidirectional packet flows with the egress SIG simply by sending the corresponding SIG frames. There is no peamble. The egress SIG, should it accept the traffic, instanciates the necessary resources on-demand to process each flow. Each such flow forms an independent sequence of packets (a stream) ordered by an incrementing sequence number. Between a given SIG ingress/egress pair, a (session ID, stream ID) pair uniquely identifies a stream.
+
+To preserve performance, IP packets encapsulated in a single stream SHOULD NOT be re-ordered. To that end:
+
+- The ingress SIG SHOULD encapsulate IP packets that cannot be proven independent (for example with the same source, destination and traffic class) in the same stream.
+- The ingress SIG SHOULD encapsulate IP packets to a given stream in the order in which they were received.
+- The ingress SIG SHOULD send all frames of a given stream over a the same SCION path.
+- The egress SIG SHOULD reassemble and forward packets from each stream, ordered by frame sequence number and packet within each frame.
```

Comment:
```suggestion
- The ingress SIG SHOULD encapsulate IP packets that cannot be proven independent (e.g., with the same  IP 6-tuple) in the same stream.
- The ingress SIG SHOULD encapsulate IP packets to a given stream in the order in which they were received.
- The ingress SIG SHOULD send all frames of a given stream over the same SCION path.
- The egress SIG SHOULD reassemble and forward packets from each stream, ordered by frame sequence number and packet within each frame.
```
We don't describe what traffic class we are referring to (the SCION header traffic class? The IP header one?). Id therefore remove the mention of traffic class